File: strings/longest_palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
      char_count = {}
      
      # Count frequency of each character
      for ch in s:
          char_count[ch] = char_count.get(ch, 0) + 1
      
      length = 0
      has_odd = False
      
      # For each character count
      for count in char_count.values():
          if count % 2 == 0:
              length += count
          else:
              length += count - 1
              has_odd = True
      
      # If we have any odd counts, we can place one character in the middle
      if has_odd:
          length += 1
      
      return length

File: strings/longest_palindrome.rb
# @param {String} s
# @return {Integer}
def longest_palindrome(s)
  char_count = {}

  s.each_char do |ch|
    char_count[ch] = (char_count[ch] || 0) + 1
  end

  length = 0
  has_odd = false

  char_count.each do |ch, count|
    if count.even?
      length += count
    else
      length += count - 1
      has_odd = true
    end
  end

  length += 1 if has_odd

  length
end